app/Models/ResourceMedia.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class ResourceMedia extends Model
+{
+    use HasFactory;
+
+    protected $table = 'resources_media';
+
+    protected $fillable = [
+        'model',
+        'path',
+    ];
```

Comment:
Si usamos relación polimórfica debemos aplicar la morpTo en el modelo de ResourceMedia
indicando que este puede pertenecer a diferentes modelos.

```suggestion
     public function model()
    {
        return $this->morphTo();
    }
    ];
```